src/battle_ai_main.c

```diff
@@ -1117,10 +1117,9 @@ static s16 AI_CheckBadMove(u8 battlerAtk, u8 battlerDef, u16 move, s16 score)
             break;
         case EFFECT_GROWTH:
         case EFFECT_ATTACK_SPATK_UP:    // work up
-            if (!BattlerStatCanRise(battlerAtk, AI_DATA->abilities[battlerAtk], STAT_ATK) || !HasMoveWithSplit(battlerAtk, SPLIT_PHYSICAL))
+            if ((!BattlerStatCanRise(battlerAtk, AI_DATA->abilities[battlerAtk], STAT_ATK) && !BattlerStatCanRise(battlerAtk, AI_DATA->abilities[battlerAtk], STAT_SPATK))
+             || (!HasMoveWithSplit(battlerAtk, SPLIT_PHYSICAL)  && !HasMoveWithSplit(battlerAtk, SPLIT_SPECIAL)))
```

Comment:
Hmm, I feel a new function is needed to check if the mon has attacking moves, to avoid looping twice.

Reply:
> Hmm, I feel a new function is needed to check if the mon has attacking moves, to avoid looping twice.

You mean a function that checks if a mon has either attacking move?

Reply:
Yeah